app/src/main/java/com/cortado/minionify/accumulators/PreferenceAccumulator.java

```diff
@@ -0,0 +1,33 @@
+package com.cortado.minionify.accumulators;
+
+import android.content.SharedPreferences;
+
+import com.cortado.minionify.mode.AppMode;
+
+public class PreferenceAccumulator implements IAccumulator {
+    private SharedPreferences mPreferences;
+
+    private static final String KEY_NOT_FOUND_DEFAULT_VALUE = " ";
+
+    public PreferenceAccumulator(SharedPreferences prefs) {
+        mPreferences = prefs;
+    }
+
+    @Override
+    public void set(String key, String value) {
+        SharedPreferences.Editor prefsEditor = mPreferences.edit();
+        prefsEditor.putString(key, value);
+        prefsEditor.commit();
+    }
+
+    @Override
+    public String get(String key) {
+        return mPreferences.getString(key, KEY_NOT_FOUND_DEFAULT_VALUE);
```

Comment:
this is bad, since you may return the default. check if exists before and throw an exception (since you expose `exists` it may be runtime)

Reply:
Done.